profiler/options.go

```diff
@@ -174,3 +189,16 @@ func WithStatsd(client StatsdClient) Option {
 		cfg.statsd = client
 	}
 }
+
+// WithSite specifies the datadog site (datadoghq.com, datadoghq.eu, etc.)
+// which profiles will be sent to.
+func WithSite(site string) Option {
+	return func(cfg *config) {
+		u, err := urlForSite(site)
+		if err != nil {
+			log.Warn("profiler: ignoring invalid site %s: %s", site, err)
```

Comment:
Maybe we should say something like `invalid site provided, using https://intake.profile.datadoghq.com/v1/input`. I would even call this a fatal error but we don't want to ever crash the main program. Should this be an `Error` instead? 🤔

Reply:
Yes, we should probably log this as an `Error`. I agree, ideally this would be a `Fatal` error, but obviously can't be. This is where we should do something like #540 for `profiler.Start()`, to be able to report configuration or startup errors.